docs/rules/no-deprecated-delete-set.md

```diff
@@ -0,0 +1,49 @@
+---
+pageClass: rule-details
+sidebarDepth: 0
+title: vue/no-deprecated-delete-set
+description: disallow using deprecated `$delete` and `$set` (in Vue.js 3.0.0+)
+---
+
+# vue/no-deprecated-delete-set
+
+> disallow using deprecated `$delete` and `$set` (in Vue.js 3.0.0+)
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> _**This rule has not been released yet.**_ </badge>
+- :gear: This rule is included in all of `"plugin:vue/vue3-essential"`, `*.configs["flat/essential"]`, `"plugin:vue/vue3-strongly-recommended"`, `*.configs["flat/strongly-recommended"]`, `"plugin:vue/vue3-recommended"` and `*.configs["flat/recommended"]`.
+
+## :book: Rule Details
+
+This rule reports use of deprecated `$delete` and `$set`. (in Vue.js 3.0.0+).
+
+<eslint-code-block :rules="{'vue/no-deprecated-delete-set': ['error']}">
+
+```vue
+<script>
+  import { set as st, delete as del } from 'vue'
+  export default {
+    mounted () {
+      /* ✗ BAD */
+      this.$set(obj, key, value)
+      this.$delete(obj, key)
+
+      Vue.set(obj, key, value)
+      Vue.delete(obj, key)
+
+      st(obj, key, value)
+      del(obj, key)
+    }
+  }
+</script>
+```
+
+</eslint-code-block>
+
+## :wrench: Options
+
+Nothing.
+
```

Comment:
Please link to <https://v3-migration.vuejs.org/breaking-changes/#removed-apis> in a `📚 Further Reading` section.